src/CoreBundle/private/library/classes/TCMSLogChange.class.php

```diff
@@ -600,6 +600,37 @@ public static function SetDisplayFieldPositionByAlias($sTableId, $sAliasField, $
         }
     }
 
+    public static function moveMenuCategoryPositionAfterTarget(
```

Comment:
How about `moveMainMenuCategory` or `setMainMenuPosition`?

Then we can rename the arguments to `mainMenuCategorySystemName` and `afterThisMainMenuCategory`. The "source" and "target" prefixes don't really fit here.

Also we should add the possibility to move a category to the begin or the end.

And please don't use boolean values to indicate an error. Throw an exception instead.

Reply:
renamed it.

Reply:
I would not throw an exception here. This is non critical but would break complete update processes if someone deleted or renamed a main menu category in his project (which will happen). But i changed it to output an update warning instead.

Reply:
OK.
Yet missing is a possibility to move a category to first position.

Reply:
I don't think we need this. Content should always be the first menu entry. It holds the basic features of the CMS that everyone needs. Last is addressed by moving after last menu entry, or not?

Reply:
If we really need the pos 1 we could add this later i would suggest.

Reply:
We had similar problems for other methods that let us set positions - we shouldn't assume that our current setup is carved in stone for all projects (if we add it later, there will be a point in time where a developer can't set the position and this will be frustrating; it's not much work to add it now I think).

Reply:
added